### PR TITLE
Adding blog post for running Unet audio on Apple Silicon

### PR DESCRIPTION
**Pull request by notthetup**

### CODE REVIEW

File: _posts/2024-02-19-running-unet-audio-on-apple-silicon.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+feature-img: "assets/img/mbp.jpg"
+thumbnail: "assets/img/mbp.jpg"
+tags: [howto, unetaudio, unetstack]
+---
+
+[Unet audio](https://unetstack.net/handbook/unet-handbook.html#_transmitting_and_recording_arbitrary_acoustic_waveforms) is a great tool for using and testing Unet implementations using UnetStack. Unet audio uses your computer's sound card as an acoustic modem. Almost the entire functionality of UnetStack is available in Unet audio, so you can easily [develop and test Unet Agents](https://unetstack.net/handbook/unet-handbook.html#_developing_your_own_agents), try out [communications algorithms](https://blog.unetstack.net/custom-phy), and even test your scripts and programs written to interact with UnetStack without needing any extra hardware.
+
+![Unet audio](/assets/img/unetaudio.png)
+
+Unet audio binds to your computer's operating system and use the built-in audio services to send and receive audio from the sound card. This means that some of the libraries used by Unet audio needs to be compiled for the specific operating system and architecture.
```
Comment (ettersi) [inline _posts/2024-02-19-running-unet-audio-on-apple-silicon.md:17]:
```suggestion
Unet audio binds to your computer's operating system and uses the built-in audio services to send and receive audio from the sound card. This means that some of the libraries used by Unet audio need to be compiled for the specific operating system and architecture.
```

File: _posts/2024-02-19-running-unet-audio-on-apple-silicon.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+Until UnetStack 3 Community release v3.4.3, Unet audio was only supported on Linux and macOS on the Intel **x86** architecture. However, with the release of Apple Silicon, and the popularity of Mac computers with the M1/2/3 chips, we have been working to make Unet audio available on the **ARM** architecture.
+
+We are happy to announce that Unet audio is now available on macOS running on Apple Silicon, from [Unet Community v3.4.4]https://unetstack.net/#downloads) onwards.
```
Comment (ettersi) [inline _posts/2024-02-19-running-unet-audio-on-apple-silicon.md:21]:
```suggestion
We are happy to announce that Unet audio is now available on macOS running on Apple Silicon, from [Unet Community v3.4.4](https://unetstack.net/#downloads) onwards.
```

File: _posts/2024-02-19-running-unet-audio-on-apple-silicon.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+Unet audio comes as a part of UnetStack 3 Community releases. You can download the latest release from the [UnetStack 3 downloads section](https://unetstack.net/#downloads).
+
+### Pre-requisites
+
+UnetStack 3 requires Java 8 to be installed on your computer. With the macOS on Apple Silicon, you can run Java 8 compiled for x86 architecture using [Rosetta 2](https://support.apple.com/en-sg/HT211861) or run Java 8 compiled for ARM (aarch64) architecture natively.
+
+Unet audio additionally requires [portaudio](https://www.portaudio.com/) to be installed on your computer. [Homebrew](https://https://brew.sh/) is a great package manager for macOS, and you can use it to install portaudio.
+
+The architecture of the Java 8 (`x86` or `aarch64`) that you have installed on your computer will determine which version of portaudio you need to use. The following table shows the compatibility between the architecture of the Mac, Java 8, and portaudio:
```
Comment (ettersi) [inline _posts/2024-02-19-running-unet-audio-on-apple-silicon.md:33]:
```suggestion
The architecture of Java 8 that you have installed on your computer will determine which version of portaudio you need to use. The following table shows the compatibility between the architecture of the Mac, Java 8, and portaudio:
```

File: _posts/2024-02-19-running-unet-audio-on-apple-silicon.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+bin/unet audio
+```
+
+## Accepting Security Permissions on macOS for running Unet audio
+
+When you run Unet audio for the first time, macOS will flag the dynamic libraries that Unet audio uses as being from an unidentified developer.
+
+![Unet audio Security](/assets/img/unetaudio-security.png){: width="50%" }
+
+You will need to be permitted to run these libraries. You can do this by going to `System Preferences` -> `Security & Privacy` -> `General` and clicking on `Allow Anyway` next to the message that says that the libraries were blocked from running.
```
Comment (ettersi) [inline _posts/2024-02-19-running-unet-audio-on-apple-silicon.md:98]:
```suggestion
You will need to permit running these libraries. You can do this by going to `System Preferences` -> `Security & Privacy` -> `General` and clicking on `Allow Anyway` next to the message that says that the libraries were blocked from running.
```

File: _posts/2024-02-19-running-unet-audio-on-apple-silicon.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+![Unet audio Security](/assets/img/unetaudio-security-allow.png){: width="70%" }
+
+When you run Unet audio again, you will be prompted to permit to run the libraries. Click on `Open` to permit to run the libraries.
```
Comment (ettersi) [inline _posts/2024-02-19-running-unet-audio-on-apple-silicon.md:102]:
```suggestion
When you run Unet audio again, you will be prompted to permit running the libraries. Click on `Open` to permit running the libraries.
```

File: _posts/2024-02-19-running-unet-audio-on-apple-silicon.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+![Unet audio Security](/assets/img/unetaudio-security-open.png){: width="50%" }
+
+You might have to run the `bin/unet audio` command again after permitting the libraries to run. You should only have to do this once, and Unet audio will run without any issues in the future.
```
Comment (ettersi) [inline _posts/2024-02-19-running-unet-audio-on-apple-silicon.md:106]:
```suggestion
You might have to run the `bin/unet audio` command again after permitting the libraries to run. You should only have to do this once, and Unet audio will run without any issues after that.
```

"in the future" could be construed as we are working on resolving this.

File: _posts/2024-02-19-running-unet-audio-on-apple-silicon.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+title: Running Unet audio on Apple silicon
+date: 14/02/2024
+author: Chinmay Pendharkar
+categories: howto
+feature-img: "assets/img/mbp.jpg"
+thumbnail: "assets/img/mbp.jpg"
+tags: [howto, unetaudio, unetstack]
+---
+
+[Unet audio](https://unetstack.net/handbook/unet-handbook.html#_transmitting_and_recording_arbitrary_acoustic_waveforms) is a great tool for using and testing Unet implementations using UnetStack. Unet audio uses your computer's sound card as an acoustic modem. Almost the entire functionality of UnetStack is available in Unet audio, so you can easily [develop and test Unet Agents](https://unetstack.net/handbook/unet-handbook.html#_developing_your_own_agents), try out [communications algorithms](https://blog.unetstack.net/custom-phy), and even test your scripts and programs written to interact with UnetStack without needing any extra hardware.
```
Comment (manuignatius) [inline _posts/2024-02-19-running-unet-audio-on-apple-silicon.md:13]:
```suggestion
[Unet audio](https://unetstack.net/handbook/unet-handbook.html#_transmitting_and_recording_arbitrary_acoustic_waveforms) is a great educational tool for using and testing Unet implementations using UnetStack. Unet audio uses your computer's sound card as an acoustic modem. Almost the entire functionality of UnetStack is available in Unet audio, so you can easily [develop and test Unet Agents](https://unetstack.net/handbook/unet-handbook.html#_developing_your_own_agents), try out [communications algorithms](https://blog.unetstack.net/custom-phy), and even test your scripts and programs written to interact with UnetStack without needing any extra hardware.
```

**Review by manuignatius — APPROVED** [review]

**Review by ettersi — APPROVED** [review]

**Pull request merged**